app/services/instrument_offline_reservation_canceler.rb

```diff
@@ -0,0 +1,44 @@
+class InstrumentOfflineReservationCanceler
+
+  def cancel!
+    reservations_to_cancel.each do |reservation|
+      reservation.transaction do
+        cancel_reservation(reservation)
+        Notifier.offline_cancellation_notification(reservation)
+      end
+    end
+  end
+
+  private
+
+  def admin
+    # OrderDetail#cancel_reservation needs an object that responds to #id
+    @admin ||= OpenStruct.new(id: 0)
+  end
+
+  def cancel_reservation(reservation)
+    reservation
+      .order_detail
+      .cancel_reservation(admin, OrderStatus.canceled.first, true, false)
+    reservation
+      .update_attribute(:canceled_reason, "The instrument was offline")
+  end
+
+  def reservations_to_cancel
+    Reservation
+      .not_offline
```

Comment:
Can we come up with a better name for this? This also excludes `AdminReservation`. Perhaps `user` or `user_reservation`? I don't love those, but I'd like to have something a little clearer.